docs/gallery/general/extensions.py

```diff
@@ -124,3 +124,113 @@ def __init__(self, *args, **kwargs):
 #     When defining your own :py:class:`~pynwb.core.NWBContainer`, the subclass name does not need to be the same
 #     as the extension type name. However, it is encouraged to keep class and extension names the same for the
 #     purposes of readibility.
+
+
+####################
+# Example: Cortical Surface Mesh
+# -----------------------------------------------------
+#
+# Here we show how to create extensions by example by creating a data class for a
+# cortical surface mesh. This data type is particularly important for ECoG data,
+# we need to know where each electrode is with respect to the gyri and sucli.
+# Surface mesh objects contain two types of data:
+#
+# 1. `vertices`, which is an (n, 3) matrix of floats that represents points in 3D space.
+#
+# 2. `faces`, which is an (n, 3) matrix of uints that represents indices of the `vertices` matrix. Each triplet of
+# points defines a triangular face, and the mesh is comprised of a collection of triangular faces.
+#
+# First, we set up our extension. I am going to use the name `ecog`
+
+from pynwb.spec import NWBDatasetSpec, NWBNamespaceBuilder, NWBGroupSpec
+
+name = 'ecog'
+ns_path = name + ".namespace.yaml"
+ext_source = name + ".extensions.yaml"
+
+####################
+# Now we define the data structures. We use :py:meth:`~pynwb.core.NWBDataInterface` as the base type,
+# which is the lowest type you are likely to use as a base. The name of the
+# class is `CorticalSurface`, and it requires two matrices, `vertices` and
+# `faces`.
+
+datasets = [NWBDatasetSpec(doc='faces for surface, indexes vertices',
+                           shape=(None, 3),
+                           name='faces', dtype='uint',
+                           dims=('face_number', 'vertex_index')),
+            NWBDatasetSpec(doc='vertices for surface, points in 3D space',
+                           shape=(None, 3),
+                           name='vertices', dtype='float',
+                           dims=('vertex_number', 'xyz'))]
+
+surface = NWBGroupSpec(doc='brain cortical surface',
+                       datasets=datasets,
+                       neurodata_type_def='CorticalSurface',
+                       neurodata_type_inc='NWBDataInterface')
+
+# Now we set up the builder and add this object
```

Comment:
add a "#######" above if you want sphinx-gallery to separate, as opposed to inline code comment